### PR TITLE
Web3.js type definitions removed

### PR DESCRIPTION
**Pull request by nivida**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by zlumer — CHANGES_REQUESTED** [review]
Can we please have some context here?

Does web3 provide their own typings from now on?
How stable are they? Were they adapted from this code or written from scratch?

As far as I understand, web3 is a JS-first project, is it really a good idea to include typings in the main repository instead of this one?
If I remember correctly, this package was born when official web3 typings became so outdated that the package became unusable from TypeScript and @Levino had to move them here.
See https://github.com/ethereum/web3.js/issues/1597, https://github.com/ethereum/web3.js/issues/1658

**Comment by Levino** [comment]
@nivida Why do you have to make this so hard? Just try to take the perspective of the reviewers. Just give them the context that they need to review and understand what is going on.

[Here is the context.](https://github.com/ethereum/web3.js/issues/1248).

TLDR: It was decided by the maintainers of `web3` to not follow best practices but provide the type definitions themselves.

There is no point in fighting this and keep the type definitions here because this will only make things harder. So I agree to remove the types from here.

**But** we have good tests here for the type definitions. Have these tests been migrated to the source of `web3` and are the test run in CI for every PR? I think this should be a requirement for moving forward. I kindly ask @nivida to provide links to the code parts of `web3` where these tests have been added to the `web3` source code.

Overall this is all very sad and very bad but there is nothing one can do about it. For the future I recommend that `web3` is forked / rewritten by others in typescript which will help the whole ecosystem.

**Comment by Levino** [comment]
https://github.com/ethereum/web3.js/issues/1248 should be closed.

**Comment by alexkvak** [comment]
@nivida web3.js typings still have the same issues that was here, e.g. #32409 and #32416. 
Have you plans to port recently fixed issues to web3.js typings?

**Comment by nivida** [comment]
> Can we please have some context here?

> Does web3 provide their own typings from now on?
> How stable are they? Were they adapted from this code or written from scratch?

> As far as I understand, web3 is a JS-first project, is it really a good idea to include typings in the main repository instead of this one?
> If I remember correctly, this package was born when official web3 typings became so outdated that > the package became unusable from TypeScript and @Levino had to move them here.
> See ethereum/web3.js#1597, ethereum/web3.js#1658

@zlumer Yes, Web3.js does now provide their own typings. The typings are stable and checked by me the maintainer of this library. I will always update, check and improve them and they are written from scratch because of a bigger refactoring I've done. 

--------

> @nivida Why do you have to make this so hard? Just try to take the perspective of the reviewers. Just give them the context that they need to review and understand what is going on.

> TLDR: It was decided by the maintainers of web3 to not follow best practices but provide the type definitions themselves.

> There is no point in fighting this and keep the type definitions here because this will only make things harder. So I agree to remove the types from here.

> But we have good tests here for the type definitions. Have these tests been migrated to the source of web3 and are the test run in CI for every PR? I think this should be a requirement for moving forward. I kindly ask @nivida to provide links to the code parts of web3 where these tests have been added to the web3 source code.

@Levino 
First of all, it was not decided by me, it was decided by the developers which are involved in the discussion. You will find in the types folder of the desired package a tests folder with the dtslint tests as you would have them here on DT. Dtslint will get executed on each PR with travis-ci. 
Example types & tests: https://github.com/ethereum/web3.js/tree/1.0/packages/web3-providers/types

> Overall this is all very sad and very bad but there is nothing one can do about it. For the future I recommend that web3 is forked / rewritten by others in typescript which will help the whole ecosystem.

I don't recommend doing it. The latest refactoring I've done is the first and is will definitely not the last. The important point is now to release a stable 1.0 version of this library before doing something else. I'm a big fan of TypeScript but changes in this direction can be done post stable release. Hopefully you will join in the discussion I've mentioned in my Medium post about the future API of the library.
Medium: https://medium.com/@samuel_91690/web3-js-v1-0-0-beta-38-8d2bb6e73d0b

--------

> @nivida web3.js typings still have the same issues that was here, e.g. #32409 and #32416.
Have you plans to port recently fixed issues to web3.js typings?

@alexkvak I'll check the issues here and will if required fixed them in the new typings.

**Comment by zlumer** [comment]
> [Here is the context..](https://github.com/ethereum/web3.js/issues/1248)

Thanks. Locking the thread mid-discussion usually means that there are unresolved problems, which is definitely the case here.

I'm actively against this deletion.
Let me explain my position and ask a few questions to the web3 team:

1. I still remember the times when web3 typings didn't work. It was horrible. It was unusable.
Look at this line: [`"postinstall": "rm -f node_modules/web3/*.d.ts"`](https://github.com/DucaturFw/ducor-backend/blob/master/package.json#L9)
Web3 typings were so bad that we had to manually delete them from the project to make it work with newer DT typings.
What's going to be different this time? Do you have any new rules for PRs that require including TS definitions before merging? Isn't it better to just migrate to TS then?

2. It's against TypeScript guidelines. [TypeScript docs](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) clearly say that unless your project is written in TypeScript itself, it's better to maintain typings in central DT repository. Lots of great TS programmers here, automatic tests/CI, contribution guidelines etc.
Are you sure you'll do a better job internally than all the DT contributors?
Which brings us to the next point:

3. Original web3 and DefinitelyTyped contributors.
Looking at the code, I can see only one author mentioned: @joshstevens19.
This means that either typings from this repo were refactored and author list was removed (which is bad), or that @joshstevens19 re-did all typings by himself (which is even worse, no offense).
In this repo we have [20 different contributors](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/web3/index.d.ts#L3-L22). Just this week 4 PRs from two different authors were merged into master.
Not trying to attack Josh personally, but do you really think that 1 great developer in a couple months can do better than 20 great developers in a year in terms of edge-cases and real-life production usage?
@alexkvak mentioned that some of the bugs that were fixed here are not yet ported to your codebase.
The decision to delete (working and tested) typings from DT looks premature.

In my opinion, the better way to improve typings was to contribute here with fixes, splitting typings for different packages etc.
In the locked thread someone said that PRs to DefinitelyTyped take a long time.
Look at the last week commits #32409, #32416, #32417, #32431 - all of them were merged on the same day or the day after. They are already published as NPM package. Are you sure you can consistently publish typings fixes from a new PR into NPM package in a day or two?

Also, I don't see any recent commits from web3 team, so not sure if this long acceptance time is personal experience or mere speculation.

I suggest that before removing `@types/web3` package, the core web3 team should:
1. Port all tests from this repository to web3.
2. Attribute all code authors from this repo in your new typings next to @joshstevens19 (where applicable).
3. Rewrite the codebase in TypeScript. Seriously. There's no point in maintaining JS code and TS typings separately, when TypeScript is basically JS with types. Just combine them!
If there are any reasons for you to not migrate to TypeScript, then these same reasons apply to moving types from DT to web3 repo. Simple as that.

**Comment by zlumer** [comment]
Sorry @nivida, haven't seen your reply until posting.

>> Overall this is all very sad and very bad but there is nothing one can do about it. For the future I recommend that web3 is forked / rewritten by others in typescript which will help the whole ecosystem.
> 
> I don't recommend doing it. The latest refactoring I've done is the first and is definitely not the last. The important point is now to release a stable 1.0 version of this library before doing something else. I'm a big fan of TypeScript but changes in this direction can be done post stable release. Hopefully you will join the discussions I've mentioned in my Medium post about the future API of the library.
> Medium: https://medium.com/@samuel_91690/web3-js-v1-0-0-beta-38-8d2bb6e73d0b

I'd argue that the refactoring (or rewriting) of definitions made by @joshstevens19 is as important to TypeScript users as migrating to TS would be for web3 team.

What me and @Levino are trying to communicate here is that if you consider TypeScript first-class citizen of your codebase, you should just migrate to it (with a simple `noImplicitAny: false` in tsconfig.json).
And if you're still deciding whether migration is worth the cost, you can let the TypeScript community handle typings for the time being. We're really good at it, because we use these typings daily in our work. PRs are always welcome of course :smiley:

**Comment by joshstevens19** [comment]
Hi @zlumer i completely understand your annoyance when you have put a lot of work in the typing's for this. Firstly this discussion started when I reached out because the typing's within here never matched up and were always out of date (sorry its true 👎 ). I work for FunFair and I am creating blockchain solutions everyday using TypeScript a lot of this became a huge pain. After lots of research I found out the Redux and Lodash had this problem and they resolved this by bringing types in-house which improved so much for their TS devs. This made sense as now your web3 version and the types can always be in sync (which was a huge problem before) and we can enforce maintaining/extending them when PRs come in. So let me try to answer your points you have raised but before i start this all of this is coming from a good place. We all want the web3 experience to be the best it can be and if this supports it I don't see how it causes any harm at all. 

> 1. I still remember the times when web3 typings didn't work. It was horrible. It was unusable.
> Look at this line: "postinstall": "rm -f node_modules/web3/*.d.ts"
> Web3 typings were so bad that we had to manually delete them from the project to make it work with newer DT typings.
> What's going to be different this time? Do you have any new rules for PRs that require including TS definitions before merging? Isn't it better to just migrate to TS then?

Yes we have updated the PR template already, we have written full tests using the same library @types use to make sure they all match the types. This was so we get greater visibility and control to make sure our types never get out of sync. I will personally make sure I look out for this and @nivida will 1000% inform this so i assure you it will not get out of date, also it is now on the PR template and is a mandatory thing. 

>  2. It's against TypeScript guidelines. TypeScript docs clearly say that unless your project is written in TypeScript itself, it's better to maintain typings in central DT repository. Lots of great TS programmers here, automatic tests/CI, contribution guidelines etc.
> Are you sure you'll do a better job internally than all the DT contributors?

We already had these discussions, typescript never says you `must` they say the word `preferred approach`, if that `preferred approach` is not working like it didn't for Redux who have 3.5M downloads a week and their TS support is 2nd to none then you can go with this way. Why can't the people who care about the types do the PRs with web3 itself? nothing changes here, we have written FULL tests using the same lib `dtslint` as `@types` does, we have enforced contribution guidelines. Nobody saying we can do a better job we are saying that this movement made sense to us and the community agreed with 78% votes they wanted it internally, how can you argue with that?

> 3. Original web3 and DefinitelyTyped contributors.
> Looking at the code, I can see only one author mentioned: @joshstevens19.
> This means that either typings from this repo were refactored and author list was removed (which is bad), or that @joshstevens19 re-did all typings by himself (which is even worse, no offense).
> In this repo we have 20 different contributors. Just this week 4 PRs from two different authors were merged into master.
> Not trying to attack Josh personally, but do you really think that 1 great developer in a couple months can do better than 20 great developers in a year in terms of edge-cases and real-life production usage?
> @alexkvak mentioned that some of the bugs that were fixed here are not yet ported to your codebase.
> The decision to delete (working and tested) typings from DT looks premature

 So i want to clear this up, I did not even look at your typings when i wrote these. I started from scratch and spent HOURS making sure they matched the API guidelines, the ACTUALLY code and with @nivida refactor. I did NOT steal anyone's code and put my name against it. I went with a clear head looking at the code, fixing any docs which were not correct and writing the types as i went along with full tests against them. I am not trying to say I am better then 20 developers, i'm saying the types now match the docs and we have enforced a way this can never get out of date. So to clear that up i have not just copied and paste. Why is redoing all the typings a bad thing? me and @nivida were in constant communication and we came up with a really nice solution fully tested which should HUGELY benefit the blockchain web3 experience. I know this first hand as I use it everyday and already this has greatly improved our types and highlighted a few issues in our codebase.  

> What me and @Levino are trying to communicate here is that if you consider TypeScript first-class citizen of your codebase, you should just migrate to it (with a simple noImplicitAny: false in tsconfig.json).
> And if you're still deciding whether migration is worth the cost, you can let the TypeScript community handle typings for the time being. We're really good at it, because we use these typings daily in our work. PRs are always welcome of course

We welcome you to do PR on web3 itself, the community strongly voted for this change so what makes you view correct over theirs? I personally would love to write web3 in ts and generate it that way but bigger priorities come above that. This is the way we can get best of both worlds and fix the constant issues with the typings, as much as you don't see it now this will greatly improve the experience for people using it in TS. Lets work together to make this library the best it can be instead of arguing over nothing really here. There is no valid reasons why it should be in `@types` and the discussion we had before which you tagged nobody could name any reasons but `ts said you should` which they give you a `preferred approach` which some of the biggest libs do not follow and our successful. 

That's all I've got to say on this and I did not want to annoy anyone, your PR are welcome don't let this stop you helping out.

**Comment by zlumer** [comment]
@joshstevens19 thank you for the detailed response. I feel the need to clarify my stance a bit further here.
Disclaimer: I own a small amount of FUN tokens and I have huge respect for your work done on web3.

I'm not at all opposed to the goals you and @nivida stated here: it's obviously a good thing to have correct and up to date type definitions that align with package versioning.

The problem is, the only way to achieve this goal is to entirely migrate web3 codebase to TypeScript.

I'll dispute mainly these points:
> Yes we have updated the PR template already, we have written full tests using the same library @types use to make sure they all match the types. This was so we get greater visibility and control to make sure our types never get out of sync. I will personally make sure I look out for this and @nivida will 1000% enforce this so i assure you it will not get out of date, also it is now on the PR template and is a mandatory thing.

> nothing changes here, we have written FULL tests using the same lib dtslint as @types does, we have enforced contribution guidelines. Nobody saying we can do a better job we are saying that this movement made sense to us and the community agreed with 78% votes they wanted it internally, how can you argue with that?

> the types now match the docs and we have enforced a way this can never get out of date

I tried to explain that in my previous post, but let's use a couple of examples to make it more clear:
Fresh:
#32409 by @alexkvak fixed incorrect `fromWei()` method signature. New web3 typings have that same bug [here](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/types/index.d.ts#L149) and [here](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/types/index.d.ts#L100)
#32416 by @alexkvak fixed incorrect `toWei()` method signature. New web3 typings still have this bug [here](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/types/index.d.ts#L119) and [here](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/types/index.d.ts#L168)
Older:
#27528 by me improved `toWei()` method signature by splitting it to different overloads based on argument types. Not implemented in [new typings](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/types/index.d.ts#L168) yet.
#29495 by me fixed `sign()` and `signTransaction()` method signatures to return correct types based on docs and web3 source code (still valid after [refactoring](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-accounts/src/Accounts.js#L144-L247)). New [typings](https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-accounts/types/index.d.ts#L33) do not contain these changes and also incorrectly miss wrapping the result in a Promise.

These are the examples I could find by glancing over recent PRs to DefinitelyTyped. How many more are there that we don't know about?

It should be pretty clear now that the typings currently bundled with web3, while definitely improving on some points, in other areas introduce regressions compared to this repo.
I feel that it's truly important to first match the `web3` typings to `@types/web3` typings and only after that consider removing `@types/web3` altogether.

Having said that, we can obviously just fix these regressions by copying tests from DefinitelyTyped to web3, make PRs for every bug we find and improve the declarations where possible. But I feel like this is basically redoing the work already done here, only on web3 repo.

Again, we are not making definitions more up to date or more correct by moving them between repositories, the only way to get rid of types mismatch is to create a `package.json` in web3 and gradually add types to the whole codebase.

I hope we can reach a consensus here.

**Comment by joshstevens19** [comment]
@zlumer Thanks for your nice response. I love that you have just done 2 PR in the web3 lib, thanks so much. I want nothing more then you talented people to fix any mistakes you see, all in all we have the same vision to make web3 have perfect typing's so coming together is a must even if we may 

`The problem is, the only way to achieve this goal is to entirely migrate web3 codebase to TypeScript.` 
I love TS and would love this but its not my call i'm afraid, it seems other things come above this. 

> 
 It should be pretty clear now that the typings currently bundled with web3, while definitely improving on some points, in other areas introduce regressions compared to this repo.

Yes the types may have a few bugs which this lib has picked up and fixed and I would love nothing more then your help to get them correct in the web3 lib!! Which i have seen you have already done. 

> These are the examples I could find by glancing over recent PRs to DefinitelyTyped. How many more are there that we don't know about?

Probably a few more but this can all be fixed with a few PRs which we would greatly appreciated. Its not going to perfect phase 1 but with peoples help we can make it perfect like Redux has done. 

> Having said that, we can obviously just fix these regressions by copying tests from DefinitelyTyped to web3, make PRs for every bug we find and improve the declarations where possible. But I feel like this is basically redoing the work already done here, only on web3 repo.

If you read my points about the different packages i think you will understand how this greatly improves everything read below:

> Web3 now has different packages, you can install web3-utils as standalone and all of them if you want. Let’s say for example you have a lower version of web3-eth-accounts installed but the latest version of web3-utils, say methods had changed in web3-eth-accounts and web3-utils across versions how does someone install the correct @types library for that situation? The versions of the @types don’t go hand in hand with the web3 package versions, the types are now defined in each package meaning when you update a single package your typings for just that package will update which you can’t achieve with a single @types lib and creating one for each would become a huge annoyance for the developer and a massive maintenance effort, this solves all of this and I think is a huge positive for this move.

I have also gave more valid points why it got moved above and in the context article. All in all it came down to a vote and the community wanted it. 

I am happy for you to put as many more tests in the web3 repo itself, or migrate anything you think is missing in it. I think we discussed this to the end now. I will finish off by saying, even though you may not agree with the move it brings benefits to the lib and its going to stay, I would love nothing more then you contributing to the web3 types and us making it the best we can. 

I will review your PRs when i get a second! Thanks a lot for doing them. I hope we have many more to come. 

p.s. anything you feel needs migrating into the web3 lib itself please do we can then discuss in a PR. 

Hope we can move on and keep your support and passion for making the typing's perfect. 

Thanks
Josh

**Comment by joshstevens19** [comment]
#29495 fixed - https://github.com/ethereum/web3.js/pull/2217
#32409 & #32416 & #27528 fixed - https://github.com/ethereum/web3.js/pull/2218

👍  @zlumer

**Comment by Levino** [comment]
I did not want to rediscuss the question of whether or not to maintain the type definitions here or in `web3` which is pointless. This discussion is over and a waste of time. In the end the maintainer**s** decide such questions. I just asked some questions and @nvidia gave more context. I think this is all we can get and this PR should be merged, if there really are tests for the types in `web3` which I did not verify yet.

Thank you for agreeing with me @zlumer. My bet is that this will all be undone one day, but only time will show.

**Comment by Levino** [comment]
The underlying problem here is the way this is all communicated: @nivida you come in here and open this very drastic pull request which **must** create a big disturbance. And instead of forseeing that this will create a lot of discussing and preactivly adding all information needed for people to see through this (context, links to the new tests, explanation that they are run in CI etc, which are all obvious questions) you just stay mute and wait for the shit to hit the fan. I guess you are in some bizarre bubble of yours where you think that the whole world reads every medium post you create and a such all of this is common knowledge but it is just not. **That** is why I think that `web3` is not going to go anywhere und the current leadership.

**Comment by Levino** [comment]
**This is just a remark. The topic itself has been decided for me**

OMG I just read your medium blogpost.

> For supporting the discussion about where they are placed, I created a poll on Twitter to let the broader community decide where the definitions should live. I saw this as a test if the community would like to vote on usability questions over twitter.

A twitter poll is not a suitable tool. Very few developers have the time to read twitter at all, me included.

**Comment by Levino** [comment]
Build fails. As one step to fix this I created https://github.com/ethereum/web3.js/issues/2226

**Comment by zlumer** [comment]
>> For supporting the discussion about where they are placed, I created a poll on Twitter to let the broader community decide where the definitions should live. I saw this as a test if the community would like to vote on usability questions over twitter.
> 
> A twitter poll is not a suitable tool. Very few developers have the time to read twitter at all, me included.

Especially when there were [26 total votes](https://twitter.com/web3_js/status/1068124530822901760) of which 20 voted to move definitions to web3 repository.
We have as much [actual credited contributors](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/web3/index.d.ts#L3-L22) here on DT.

**Comment by zlumer** [comment]
> joshstevens19:
> #29495 fixed - ethereum/web3.js#2217
> #32409 & #32416 & #27528 fixed - ethereum/web3.js#2218

You haven't credited anywhere in the code the actual PR author @alexkvak who found these bugs and provided the fix for them that made it into your repository (even if indirectly).

**Comment by joshstevens19** [comment]
Lol @zlumer I did the fixes and did the PR so why would I credit someone else’s name ha ha, happy for anyone else who adds to types to add their name. I think your picking at straws now it’s getting a bit pointless. I fixed them very quick and a deployment happened today so maybe a thanks for the update would of been the best approach. 

Hopefully we see you guys do some PR in the future. Take care 👍

**Comment by zlumer** [comment]
> Lol @zlumer I did the fixes and did the PR so why would I credit someone else’s name ha ha

That basically sums it all up.

**Comment by joshstevens19** [comment]
Your trying to argue for no reason, the fix was 1 line and I pulled the branch, wrote the tests submitted the PR, sorry I did not think about putting someone’s name above. Jeeeze I have seen how you enjoy arguing with mostly everyone in your comments so hey you succeed again. I have approached this all very nice but you just kept going on the attack. This conversation is basically done now anyway.  

If you want to migrate  anything from here into the web3 I will happily look at your PRs, if not can we approve the closure of it please. 

Cheers
Josh

**Comment by Levino** [comment]
@zlumer there really is no point to this. All we can do is to support the best we can without expecting much. I suggest to continue any discussion at `web3` and merge this here and get it over with.

@nvidia please make the build pass

**Review by phra — APPROVED** [review]

**Comment by alexkvak** [comment]
@nivida cannot find typings for `web3.eth.personal.importRawKey` in web3.js repo, but it presents here. Have you plan to migrate all presented here typings into web3.js repo?

**Comment by joshstevens19** [comment]
Hey @alexkvak `web3-eth-personal` was one of the only typing's i did not do. Looking at the typing's it was missing a few which i have done a PR for here and should be in the next release, thanks for pointing it out. 

PR:
https://github.com/ethereum/web3.js/pull/2278

In terms of migrate all presented typing's in this repo, we do not plan to do that. We have written the typing's from scratch going against the code itself and updating the incorrect documentation as we went along (as we found a lot was incorrect). We highly promote if you find any incorrect/missing typing's just do a PR in the `web3.js` lib itself. This repo will be closed once @nivida fixes his PR. 

Cheers look forward to some PR from you in the `web3` lib 👍

**Comment by Levino** [comment]
> In terms of migrate all presented typing's in this repo, we do not plan to do that.

Okay. But you should migrate all tests from here to the `web3` repo. Don't you think? More tests is always better, right?

**Comment by joshstevens19** [comment]
We have written full test coverage on all the types which now is mandatory to pass else the entire build will fail. We use the same `dstlint` package (which i had to do a PR to fix) that `DT` executes when running your tests. I don't see the need in repeating tests but if you run through what you have against what we have in `web3` lib and see any you feel is needed please add them and do a PR. I have had a quick look through and it seems all your tests you have we have covered in the `web3` lib (files i reviewed in your code were `web3-tests.ts` and `abi-tests.ts`).

FYI all tests are in the `types/tests` folder in the package itself on the `web3` lib. 

👍

**Comment by Levino** [comment]
A simple "no" is also fine. Thank you.

**Comment by joshstevens19** [comment]
@Levino I like to explain my point rather then come across rude by accident. Thanks Levino 👍

**Comment by Levino** [comment]
To be honest: You come accross quite rude and ignorant anyhow.

**Comment by joshstevens19** [comment]
wow lol I have only been welcoming and honest with my responses. Ditto back at you. 

If you think something needs changing do a PR yourself! simple as that. 👍

**Comment by Levino** [comment]
Please make the build pass. Please @nvidia

**Comment by nivida** [comment]
@Levino Will do it on monday I'm currently sick in the bed.

**Comment by Levino** [comment]
Gute Besserung
